src/transaction_signature_options/mod.rs

```diff
@@ -119,96 +119,160 @@ impl interactive_clap::FromCli for Submit {
         }
 
         match optional_clap_variant {
-            Some(CliSubmit::Send) => {
-                match (context.on_before_sending_transaction_callback)(
-                    &context.signed_transaction,
-                    &context.network_config,
-                    &mut storage_message,
-                ) {
-                    Ok(_) => (),
-                    Err(report) => {
-                        return interactive_clap::ResultFromCli::Err(
-                            optional_clap_variant,
-                            color_eyre::Report::msg(report),
-                        )
-                    }
-                };
+            Some(CliSubmit::Send) => match context.signed_transaction {
+                SignedTransactionOrSignedDelegateAction::SignedTransaction(signed_transaction) => {
+                    match (context.on_before_sending_transaction_callback)(
+                        &signed_transaction,
+                        &context.network_config,
+                        &mut storage_message,
+                    ) {
+                        Ok(_) => (),
+                        Err(report) => {
+                            return interactive_clap::ResultFromCli::Err(
+                                optional_clap_variant,
+                                color_eyre::Report::msg(report),
+                            )
+                        }
+                    };
```

Comment:
Use `if let` here and in similar places below to reduce the nesting and make it easier to read the code.

```rust
if let Err(report) = callback() {
}
```